src/main/java/cloud/orbit/spring/actuate/ActorInfoContributorLifetimeExtension.java

```diff
@@ -0,0 +1,176 @@
+
+    ActorInfoContributorLifetimeExtension(final OrbitRemoteReferenceService orbitRemoteReferenceService,
+                                          final LifetimeExtension springLifetimeExtension)
+    {
+        this.orbitRemoteReferenceService = orbitRemoteReferenceService;
+        this.springLifetimeExtension = springLifetimeExtension;
+    }
+
+    @Override
+    public <T> T newInstance(final Class<T> concreteClass)
+    {
+        // In case Orbit chooses this extension as the "primary" LifetimeExtension, we need to redirect to the
+        // SpringLifetimeExtension for constructing new instances
+        return springLifetimeExtension.newInstance(concreteClass);
+    }
+
+    @Override
+    public Task<?> preActivation(final AbstractActor<?> actor)
+    {
+        return doIfInfoContributor(actor, actorInfoContributorReferences::add);
+    }
+
+    @Override
+    public Task<?> preDeactivation(final AbstractActor<?> actor)
+    {
+        return doIfInfoContributor(actor, actorInfoContributorReferences::remove);
+    }
+
+    private Task<?> doIfInfoContributor(final AbstractActor<?> actor,
+                                        final Consumer<ActorInfoContributorReference> synchronizedAction)
+    {
+        if (actor instanceof InfoContributor)
+        {
+            ActorInfoContributorReference actorInfoContributorReference = new ActorInfoContributorReference(actor);
+            synchronized (actorInfoContributorReferences)
+            {
+                synchronizedAction.accept(actorInfoContributorReference);
+            }
+        }
+        return Task.done();
+    }
+
+    @Override
+    public void contribute(final Info.Builder builder)
+    {
+        createActorInfoObject().ifPresent(actorInfoObject -> builder.withDetail("actors", actorInfoObject));
+    }
+
+    private Optional<?> createActorInfoObject()
+    {
+        Set<ActorInfoContributorReference> actorInfoContributorReferencesCopy;
+        synchronized (actorInfoContributorReferences)
+        {
+            actorInfoContributorReferencesCopy = actorInfoContributorReferences.stream()
+                    .map(ActorInfoContributorReference::new)
+                    .collect(Collectors.toSet());
+        }
+        Collection<ActorInfoContributorReference> expiredReferences = new ArrayList<>();
+        Map<String, Map<String, Object>> actorInfoObject = new HashMap<>();
+        actorInfoContributorReferencesCopy.forEach(actor ->
+        {
+            InfoContributor contributor = actor.reference.get();
+            if (contributor == null)
+            {
+                expiredReferences.add(actor);
+            }
+            else
+            {
+                final Info.Builder builder = new Info.Builder();
+                contributor.contribute(builder);
+                actorInfoObject.putIfAbsent(actor.name, new HashMap<>());
+                actorInfoObject.get(actor.name).put(actor.identity, builder.build().getDetails());
```

Comment:
For a use case scenario, imagine there is an actor type whose responsibility is to process a large set of data by paging through a database. The page offset might not be stored in the actor state; it might be an in-memory field. There would be several of these actors distributed across a cluster for more efficient processing, each with a different identity string.

Now, an ops person needs to debug why a chunk of data was skipped during processing. Maybe the actor is down, or maybe it got stuck on a particular page. It would be useful to be able to hit the built-in `/info` endpoint to understand the status of the actors doing this processing work.

So, in a case where you have a specific actor type you're interested in, and the info itself isn't necessarily stored in actor state, the Spring Actuator `/info` endpoint is a sensible place to look for this info. It would need to be broken down by actor identity since each actor instance would have a different page value in this example.

I would like to avoid adding a new management endpoint if possible if there is an existing one that is well-known.

That said, I am not entirely opposed to adding an endpoint that does something _different_ than what `/info` is typically used for. I believe you are describing an `/actors` endpoint that reveals the actor state for each actor on a node? I think this might be out of the scope of the `orbit-spring` module, which as I understand exists to bridge the gap between the Orbit framework and the Spring framework, not to add new features altogether. I would justify the `/info` structure that I proposed as a way of utilizing an existing Spring module and fitting it to the Orbit structure.